cylc/flow/time_parser.py

```diff
@@ -355,78 +358,69 @@ def _get_interval_from_expression(
     def _get_min_from_expression(
         self,
         expr: str,
-        context: 'TimePoint'
+        context: Optional['TimePoint']
     ) -> str:
         points: List[str] = [
             x.strip() for x in re.findall(self.MIN_REGEX, expr)[0].split(",")
         ]
         ptslist: List['TimePoint'] = []
         min_entry = ""
         for point in points:
-            cpoint, offset = self._get_point_from_expression(
-                point, context, allow_truncated=True)
+            cpoint, offsets = self._get_point_from_expression(
+                point, context, allow_truncated=True
+            )
             if cpoint is not None:
                 if cpoint.truncated:
-                    cpoint += context
-                if offset is not None:
+                    cpoint += context  # type: ignore[operator]
+                for offset in offsets:
                     cpoint += offset
-            ptslist.append(cpoint)
-            if cpoint == min(ptslist):
-                min_entry = point
+                ptslist.append(cpoint)
+                if cpoint == min(ptslist):
+                    min_entry = point
```

Comment:
```python
>>> min([1, None])
Traceback ...
TypeError
```

How has this one not blown up before?

Hole in the tests?

Reply:
Not sure. Perhaps it is very unlikely to be hit. It's difficult to tell because the `self._get_point_from_expression()` method probably does too much for a single method.

As for Mypy not flagging the appending `None` to `List[TimePoint]`, I think that might be due to the fact we're missing `py.typed` in isodatetime (added in https://github.com/metomi/isodatetime/pull/203)